src/session-pool.ts

```diff
@@ -712,7 +713,7 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
       throw new Error('No resources available.');
     }
 
-    let removeListener;
+    let removeListener: Function|undefined;
```

Comment:
I think if we're calling this function with `!` we could probably just set the type to `Function`.